src/core/server/opensearch/version_check/ensure_opensearch_version.ts

```diff
@@ -36,16 +36,57 @@
  */
 
 import { timer, of, from, Observable } from 'rxjs';
-import { map, distinctUntilChanged, catchError, exhaustMap } from 'rxjs/operators';
+import { map, distinctUntilChanged, catchError, exhaustMap, mergeMap } from 'rxjs/operators';
+import { get, forEach } from 'lodash';
 import {
   opensearchVersionCompatibleWithOpenSearchDashboards,
   opensearchVersionEqualsOpenSearchDashboards,
 } from './opensearch_opensearch_dashboards_version_compatability';
 import { Logger } from '../../logging';
 import type { OpenSearchClient } from '../client';
 
+/**
+ * Checks if all nodes in the cluster have the same cluster id node attribute
+ * that is supplied through the healthcheck param. This node attribute is configurable
+ * in opensearch_dashboards.yml.
+ * If all nodes have the same cluster id then we do not fan out the healthcheck and use '_local' node
+ * If there are multiple cluster ids then we use the default fan out behavior
+ * If the supplied node attribute is missing then we use a default cluster id and increment on healthcheck
+ * @param {OpenSearchClient} internalClient
+ * @param {string} healthcheck
+ * @returns {string|null} '_local' if all nodes have the same cluster_id, otherwise null
+ */
```

Comment:
Nice! Thanks for adding this. 

Minor: `healthcheck` is a bit vague. Can we opt for something more descriptive like `healthcheckAttributeName` or `healthCheckAttribute`?

Reply:
Will update